app/controllers/students_controller.rb

```diff
@@ -31,7 +31,12 @@ def index
   end
 
   def show
-    @student = Student.find(params[:id])
+    @student = Student.find_by_id(params[:id])
+
+    if !@student
+      flash[:error] = "Student not found."
+      redirect_to students_path
```

Comment:
Is there a reason we would redirect on a missing student instead of show a 404 page?